Common/Migration/Phase1/PreProcessors/ClassificationNodesPreProcessor.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.TeamFoundation.WorkItemTracking.WebApi.Models;
+using Common.Config;
+using Logging;
+using Microsoft.TeamFoundation.TestManagement.WebApi;
+
+namespace Common.Migration
+{
+
+    public class ClassificationNodesPreProcessor : IPhase1PreProcessor
+    {
+        static ILogger Logger { get; } = MigratorLogging.CreateLogger<ClassificationNodesPreProcessor>();
+        private IMigrationContext context;
+
+        public string Name => "Classification Nodes (Area Paths/Iterations)";
+
+        public bool IsEnabled(ConfigJson config)
+        {
+            return true;
+        }
+
+        public async Task Prepare(IMigrationContext context)
+        {
+            this.context = context;
+        }
+
+        public async Task Process(IBatchMigrationContext batchContext)
+        {
+            int modificationCount = 0;
+            if (context.Config.MoveAreaPaths || context.Config.MoveIterations)
+            {
+                await Task.WhenAll(
+                    Migrator.ReadSourceNodes(context, context.Config.SourceConnection.Project),
+                    Migrator.ReadTargetNodes(context, context.Config.TargetConnection.Project));
+            }
+
+            #region Process area paths ..
+            if (context.Config.MoveAreaPaths)
+            {
+                modificationCount = await ProcessAreaPaths(batchContext, modificationCount);
+            }
+            #endregion
+
+            #region Process iterations ..
+            if (context.Config.MoveIterations)
+            {
+                modificationCount = await ProcessIterationPaths(batchContext, modificationCount);
+            }
+            #endregion
+
+            if (modificationCount > 0)
```

Comment:
i'm not a fan of the pattern of passing an int and then assigning it in the return.  Either make it an out or just have two variables.